compiler/AST/ImportStmt.cpp

```diff
@@ -394,6 +376,66 @@ void ImportStmt::validateUnqualified() {
   }
 }
 
+/************************************* | **************************************
+*                                                                             *
+* Determine if the provided type was named in some form in the import         *
+* statement.  Used for determining if we should traverse the import statement *
+* to find its methods.                                                        *
+*                                                                             *
+************************************** | *************************************/
+std::set<const char*> ImportStmt::typeWasNamed(Type* t) const {
+  std::set<const char*> namedTypes;
+
+  typeWasNamed(t, &namedTypes);
+  return namedTypes;
+}
+
+void ImportStmt::typeWasNamed(Type* t,
+                              std::set<const char*>* namedTypes) const {
+  // We don't define any symbols for unqualified access, so the type was not
+  // listed
+  if (!providesUnqualifiedAccess()) {
+    return;
+  } else {
+    std::string nameToCheck = t->symbol->name;
+    if (AggregateType* at = toAggregateType(t)) {
+      if (at->instantiatedFrom != NULL) {
+        // Need to check against the generic type's name rather than the
+        // instantiation, since the instantiation name included instantiation
+        // information in it (and that isn't usable in a use/import list, at
+        // least right now)
+        AggregateType* rootType = at->getRootInstantiation();
+        nameToCheck = rootType->symbol->name;
+      }
+    }
+
+    const char* name = astr(nameToCheck.c_str());
+    // Otherwise, look through the list of unqualified symbol names to see if
+    // this one was listed
+    for_vector(const char, toCheck, unqualified) {
+      if (astr(toCheck) == name)
+        namedTypes->insert(astr(toCheck));
+    }
+
+    // Including if it was renamed
+    for(std::map<const char*, const char*>::const_iterator it = renamed.begin();
+        it != renamed.end();
+        ++it) {
+      if (name == astr(it->first)) {
+        // Save the original name because we'll be looking in its scope
+        namedTypes->insert(astr(it->second));
+      }
+    }
+
+    // If a parent type was named, we want to traverse this import statement
+    if (AggregateType* at = toAggregateType(t)) {
```

Comment:
I don't think you should have this code (see my questions on the tests)

Reply:
For posterity, we clarified what was going on here, so this comment is no longer relevant